### DISCUSSION TITLE
Results object doesn't have attribute 'show'

### DISCUSSION THREAD

**Discussion by nicholastu2** [ultralytics/ultralytics#9125] [Q&A]
Hello, I'm trying to use YOLO for image segmentation, so I ran the following code to do so:

`from PIL import Image
from ultralytics import YOLO

model = YOLO('yolov8n-seg.pt')
results = model.train(data="/path/to/.yaml")

results = model(["img1.bmp","img2.bmp"])

for result in results:
    result.show()
    result.save(filename='result.jpg')`

but then I get the error: AttributeError: 'Results' object has no attribute 'show'. See valid attributes below.

I'm confused because at this link: https://docs.ultralytics.com/modes/predict/#key-features-of-predict-mode the documentation says that I should be able to use the show() command with a Results object. Why can't I use the show() function?

**Comment by pderrenger** [comment]
Hi there! 😊 It looks like there's a small mix-up in the usage of the `model.train()` and `model()` calls. The `result.show()` and `result.save(filename='result.jpg')` methods are indeed available for `Results` objects when you are using the model for inference or prediction, not directly after the `train` method.

From your code snippet, it seems you're attempting to visualize and save results directly after training, which isn't the typical use-case for these methods. To use `.show()` and `.save()`, you'd need to be in the context of running predictions. Here's a correct way to use `.show()`:

```python
from ultralytics import YOLO

# Load trained segmentation model
model = YOLO('yolov8n-seg.pt')

# Run inference (prediction) on images
results = model(['img1.bmp', 'img2.bmp'])

# Now you can use .show() on each result
for result in results:
    result.show()
    result.save(filename='result.jpg')
```

This adjustment ensures you're calling `.show()` on a `Results` object obtained from making predictions, aligning with the documentation you referred to. If you're still training and want to visualize training results, consider using TensorBoard during or after your training process for a comprehensive view of your model's performance and outputs. 

Hope this clears up the confusion! 🌟

**Reply by stdio159** [reply]
This still doesn't work.

**Reply by pderrenger** [reply]
@stdio159 hi there! 😊

Thank you for your patience. To assist you better, could you please provide a minimum reproducible code example? This will help us understand the issue more clearly and investigate it effectively. You can refer to our guide on creating a minimum reproducible example here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Additionally, please ensure that you are using the latest versions of `torch` and `ultralytics`. You can upgrade your packages using the following commands:

```bash
pip install --upgrade torch ultralytics
```

Once you've updated and provided the reproducible code, we'll be able to dive deeper into the issue. Thank you for your cooperation! 🌟
